de-lib/src/main/java/org/iplantc/de/diskResource/client/views/search/MetadataTermSearchField.java

```diff
@@ -0,0 +1,156 @@
+
+    public MetadataTermSearchField(OntologyAutoBeanFactory factory,
+                                   OntologyLookupServiceProxy searchProxy,
+                                   MetadataTermSearchFieldAppearance appearance) {
+        this.factory = factory;
+
+        ListStore<OntologyLookupServiceDoc> store = new ListStore<>(OntologyLookupServiceDoc::getId);
+        store.addSortInfo(new Store.StoreSortInfo<>(Comparator.comparing(OntologyLookupServiceDoc::getLabel), SortDir.ASC));
+
+        createView(store, appearance);
+
+        createCombo(store, createLoader(searchProxy, store));
+    }
+
+    private void createView(ListStore<OntologyLookupServiceDoc> store, MetadataTermSearchFieldAppearance appearance) {
+        view = new ListView<>(store, new IdentityValueProvider<>());
+        view.setCell(new AbstractCell<OntologyLookupServiceDoc>() {
+            @Override
+            public void render(Context context, OntologyLookupServiceDoc value, SafeHtmlBuilder sb) {
+                appearance.render(context, value, sb);
+            }
+        });
+    }
+
+    private PagingLoader<OntologyLookupServiceLoadConfig, PagingLoadResult<OntologyLookupServiceDoc>> createLoader(
+            OntologyLookupServiceProxy searchProxy,
+            ListStore<OntologyLookupServiceDoc> store) {
+        PagingLoader<OntologyLookupServiceLoadConfig, PagingLoadResult<OntologyLookupServiceDoc>> loader =
+                new PagingLoader<>(searchProxy);
+
+        loader.useLoadConfig(new OntologyLookupServiceLoadConfig());
+        loader.addLoadHandler(new LoadResultListStoreBinding<>(store));
+
+        loader.addBeforeLoadHandler(event -> {
+            String query = combo.getText();
+
+            if (!Strings.isNullOrEmpty(query)) {
+                event.getLoadConfig().setQuery(query);
+            }
+        });
+
+        return loader;
+    }
+
+    private void createCombo(ListStore<OntologyLookupServiceDoc> store,
+                             PagingLoader<OntologyLookupServiceLoadConfig, PagingLoadResult<OntologyLookupServiceDoc>> loader) {
+        combo = new ComboBox<>(new FreeTextComboBoxCell(store, view));
+        combo.setLoader(loader);
+        combo.setMinChars(3);
+        combo.setWidth(250);
+        combo.setHideTrigger(true);
+        combo.setForceSelection(false);
+        combo.getCell().setPageSize(loader.getLastLoadConfig().getLimit());
+    }
+
+    @Override
+    public String convertFieldValue(OntologyLookupServiceDoc object) {
+        return object == null ? null : object.getLabel();
+    }
+
+    @Override
+    public OntologyLookupServiceDoc convertModelValue(String label) {
+        OntologyLookupServiceDoc customLabel = factory.getOntologyLookupServiceDoc().as();
+        customLabel.setLabel(label);
+
+        return customLabel;
+    }
+
+    public ComboBox<OntologyLookupServiceDoc> asField() {
+        return combo;
+    }
+
+    public void setViewDebugId(String debugId) {
```

Comment:
I might have missed it but I don't think the static IDs are actually being set.

Reply:
Good point. I borrowed this method from the [UserSearchField](https://github.com/cyverse-de/ui/blob/032ca1e87c97a584c3f4f0235196db6e5a19118d/de-lib/src/main/java/org/iplantc/de/collaborators/client/util/UserSearchField.java#L248-L250).

This `MetadataTermSearchField` is only in the Metadata Template form currently, and I don't see anything else in that form getting a debug ID.

So I'll leave this method for future use.